app/src/main/java/swati4star/createpdf/util/PDFUtils.java

```diff
@@ -143,5 +145,101 @@ public void onClick(DialogInterface dialogInterface, int i) {
+        TextView text = dialog.getCustomView().findViewById(R.id.enter_password);
+        text.setText(R.string.decrypt_message);
+        mPasswordInput.addTextChangedListener(
+                new TextWatcher() {
+                    @Override
+                    public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+                    }
+
+                    @Override
+                    public void onTextChanged(CharSequence s, int start, int before, int count) {
+                        mPositiveAction.setEnabled(s.toString().trim().length() > 0);
+                    }
 
+                    @Override
+                    public void afterTextChanged(Editable input) {
+                        input_password[0] = input.toString();
+                    }
+                });
+        dialog.show();
+        mPositiveAction.setEnabled(false);
+        mPositiveAction.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                String finalOutputFile;
+
+                PdfReader reader = null;
+                try {
+                    reader = new PdfReader(file, mContext.getString(R.string.app_name).getBytes());
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+                byte[] password = new byte[0];
+                finalOutputFile = file.replace(".pdf", mContext.getString(R.string.decrypted_file));
+                password = reader.computeUserPassword();
+                byte[] input = input_password[0].getBytes();
+                Log.e("log", "INPUT : " + Arrays.toString(input));
+                Log.e("Log", "PAS : " + Arrays.toString(password));
+                Log.e("log", "Error new  "+finalOutputFile);
+                if (Arrays.equals(input, password)) {
+                    Log.e("log","File : "+file);
+                    PdfStamper stamper = null;
+                    try {
+                        stamper = new PdfStamper(reader, new FileOutputStream(finalOutputFile));
+                    } catch (DocumentException | IOException e) {
+                        e.printStackTrace();
+                    }
+                    try {
+                        stamper.close();
+                    } catch (DocumentException | IOException e) {
+                        e.printStackTrace();
+                    }
+                    Snackbar.make(Objects.requireNonNull(mContext).findViewById(android.R.id.content),
+                            R.string.password_remove,
+                            Snackbar.LENGTH_LONG).show();
+                    reader.close();
+                    dialog.dismiss();
+                    dataSetChanged.updateDataset();
+                } else {
+                    Snackbar.make(Objects.requireNonNull(mContext).findViewById(android.R.id.content),
+                            R.string.incorrect_passowrd,
```

Comment:
Make snackbar a separate function and pass relevant string instead of writing the whole piece of code again